### ISSUE TITLE
`talosctl cluster create` doesn't create distinct clusters

### ISSUE THREAD

**Issue by flokli** [talos-systems/talos#4858]
## Bug Report
I spinned up two different qemu-powered talos clusters, using `talosctl cluster create --wait --provisioner qemu [--name foo]`, and realized the second one didn't come up.

It looks like it's trying to reuse the IPs/CNI of the first one's loadbalancer.

### Environment

- Talos version: 0.14.1
- Kubernetes version: N/A
- Platform: amd64

**Comment by smira** [comment]
Please use different `--cidr` options to run two clusters in parallel.

**Comment by flokli** [comment]
Could we detect this when spinning up the CNI bundle?

The IP should already exist, and something listen on it, so we probably shouldn't continue invoking qemu if apparently something else is using that CIDR.

**Comment by flokli** [comment]
Similar to this, while iterating on https://github.com/talos-systems/talos/pull/4860, there was a lot of manual cleanup to do. Leftover `talos*` and `veth*` interfaces. Are those cleaned up in error cases?

**Comment by smira** [comment]
> Could we detect this when spinning up the CNI bundle?
> 
> The IP should already exist, and something listen on it, so we probably shouldn't continue invoking qemu if apparently something else is using that CIDR.

it's definitely something nice to have, but might be not trivial, we could check at least presence of the link on the host with the same CIDR

**Comment by flokli** [comment]
Shouldn't it be possible to just add a check to `pkg/provision/providers/vm`'s `Provisioner.CreateNetwork()`, and let it fail if there's already an interface with one of the CIDRs?

That way, `pkg/provision/providers/qemu`'s `provisioner.Create()` shouldn't start the load balancer, dhcpd, master and worker nodes in first place.

**Comment by smira** [comment]
yes, it should be possible. we just never had time to do that, but that would be definitely nice to have
